File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
class ResistorColorDuo {
      private static final String[] COLORS = {
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    };

    int value(String[] colors) {
        int firstDigit = getColorCode(colors[0].toLowerCase());
        int secondDigit = getColorCode(colors[1].toLowerCase());
        return firstDigit * 10 + secondDigit;
    }

    private int getColorCode(String color) {
        for (int i = 0; i < COLORS.length; i++) {
            if (COLORS[i].equals(color)) {
                return i;
            }
        }
        throw new IllegalArgumentException("Invalid color: " + color);
    }
}

File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export const COLORS: string[] = [
  "black", "brown", "red", "orange", "yellow", 
  "green", "blue", "violet", "grey", "white"
];

export function getColorCode(color: string): number {
  const index = COLORS.indexOf(color);
  if (index !== -1) {
    return index;
  }
  throw new Error(`Unknown color: ${color}`);
}

export function decodedValue(colors: string[]): number {
  const firstDigit = getColorCode(colors[0].toLowerCase());
  const secondDigit = getColorCode(colors[1].toLowerCase());
  return firstDigit * 10 + secondDigit;
}

